src/main/java/com/unb/tracker/web/CourseController.java

```diff
@@ -220,4 +232,24 @@ public String deleteCourse(@PathVariable Long courseId, Principal principal) {
         return "success";
     }
 
-}
+
+    @GetMapping(value = "/courses/{courseId}/export")
+    @ResponseBody
+    public String exportAttendance(@PathVariable Long courseId, ModelMap map, Principal principal, HttpServletResponse response) {
+        StringWriter writer = new StringWriter();
+        CSVWriter csvWriter = new CSVWriter(writer);
+
+        Iterable<Absence> absences = absenceRepository.findByCourseId(courseId);
+        DateFormat df = new SimpleDateFormat("MM/dd/yyyy");
+        Course course = courseRepository.findById(courseId);
+
+        List<String[]> absenceStrings = new ArrayList<>();
+        for (Absence absence : absences) {
+            absenceStrings.add(new String[]{absence.getStudent().getUsername(), df.format(absence.getDate())});
+        }
+
+        response.setHeader("Content-Disposition", "attachment; filename=" + course.getName() + "_" + course.getSection() + ".csv");
+        csvWriter.writeAll(absenceStrings);
+        return writer.toString();
```

Comment:
![image](https://user-images.githubusercontent.com/18077531/33087052-dd3686a2-cebf-11e7-9365-f1fc3f5074e7.png)

The exported csv file doesn't seem to open well in excel

Reply:
Yeah you're totally right! This isn't a problem